service/src/LevelZero.cpp

```diff
@@ -524,6 +538,425 @@ namespace geopm
+                   uint32_t num_metric = 0;
+                   ze_result = zetMetricGet(metric_group_handle.at(metric_group_idx), &num_metric, nullptr );
+                   check_ze_result(ze_result, GEOPM_ERROR_RUNTIME,
+                                   "LevelZero::" + std::string(__func__) +
+                                   ": LevelZero Metric Count query failed",
+                                   __LINE__);
+
+                   //Cache compute basic number of metrics
+                   m_devices.at(device_idx).subdevice.num_metric.push_back(num_metric);
+
+                   //Build metric map
+                   for (unsigned int metric_idx = 0; metric_idx < num_metric; metric_idx++)
+                   {
+
+                       std::vector<zet_metric_handle_t> metric_handle(num_metric);
+                       ze_result = zetMetricGet(metric_group_handle.at(metric_group_idx),
+                                                &num_metric, metric_handle.data());
+
+                       check_ze_result(ze_result, GEOPM_ERROR_RUNTIME,
+                                       "LevelZero::" + std::string(__func__) +
+                                       ": LevelZero Metric handle acquisition failed",
+                                       __LINE__);
+                       zet_metric_properties_t metric_properties;
+                       ze_result = zetMetricGetProperties(metric_handle.at(metric_idx), &metric_properties);
+
+                       check_ze_result(ze_result, GEOPM_ERROR_RUNTIME,
+                                       "LevelZero::" + std::string(__func__) +
+                                       ": LevelZero Metric Property acquisition failed",
+                                       __LINE__);
+
+                       std::string metric_name (metric_properties.name);
+
+                       //m_devices is a std::vector of structs, indexed by GPU
+                       // subdevice is a struct inside of it.  One per GPU
+                       //  metric_data is a vector, indexed by CHIP
+                       //    it contains a map of metric name (string) -> report data for N reports (vector<double>)
+                       if (m_devices.at(device_idx).subdevice.metric_data.at(subdevice_idx).count(metric_name) != 0) {
+                           throw Exception("LevelZero::" + std::string(__func__) +
+                                           ": Metric group has two metrics with the same name",
+                                           GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+                       }
+                       m_devices.at(device_idx).subdevice.metric_data.at(subdevice_idx)[metric_name] = {};
+                       m_devices.at(device_idx).subdevice.metric_data.at(subdevice_idx)["NUM_REPORTS"] = {};
+                   }
+                   // Break out of loop once we've found the group of interest
+                   break;
```

Comment:
```suggestion
```
This ```break``` is inside the if() block that starts on line 612.  I don't believe this was intentional.

Reply:
The intent is to break out of the for loop surrounding the if once we fine the ComputeBasics group that allows for ZET_METRIC_GROUP_SAMPLING_TYPE_FLAG_TIME_BASED sampling